dumper-integration-tests/redshift-tests/src/main/java/com/google/edwmigration/dumper/csv/CsvUtil.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.edwmigration.dumper.csv;
+
+import static com.google.edwmigration.dumper.base.TestConstants.TRAILING_SPACES_REGEX;
+import static java.lang.Integer.parseInt;
+
+/** A helper class for reading and extracting data from CSV files. */
+public final class CsvUtil {
+
+  private CsvUtil() {}
+
+  /**
+   * @return String or an empty string if null.
+   */
+  public static String getStringNotNull(String value) {
+    return value == null ? "" : TRAILING_SPACES_REGEX.matcher(value).replaceFirst("");
```

Comment:
Why does this method exist? It seems to be modifying the CSV data as written, on the assumption that the trailing spaces are not significant. However, the significance of the trailing spaces would seem to depend on whether the trailing spaces were enclosed within quotation marks in the original CSV data, and that's not visible to this method (too late in the chain). It should be handled by the CSV parser, which should have a configuration flag to tell it whether UNQUOTED trailing spaces are significant or not.

So, to me, this method is a caution flag because it manipulates data in potentially unsafe ways. Do we have an argument that this is semantically correct for our use case?

Reply:
A valid question.
When [dwh-assessment-extraction-tool](https://github.com/google/dwh-assessment-extraction-tool) extracts Varchar and Char fields it trims any trailing spaces, so, did we in our test framework there.
This line just inherits previous behavior.

Done.